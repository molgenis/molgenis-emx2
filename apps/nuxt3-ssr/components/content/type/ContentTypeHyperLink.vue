<script setup lang="ts">
import { ISectionField } from "~/interfaces/types";

defineProps<{
  field: ISectionField;
}>();
</script>

<template>
  <HyperLink :href="field.value" target="_blank"></HyperLink>
</template>
