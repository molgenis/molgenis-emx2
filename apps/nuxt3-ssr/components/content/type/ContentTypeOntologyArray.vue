<script setup lang="ts">
import { ISectionField, IOntologyNode } from "~/interfaces/types";
const { field } = defineProps<{
  field: ISectionField;
}>();

const asString = computed(() => {
  return field.value?.map((item): any => item.name).join(", ");
});
</script>

<template>
  <p>{{ asString }}</p>
</template>
