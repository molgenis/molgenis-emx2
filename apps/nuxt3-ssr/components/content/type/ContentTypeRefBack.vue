<script setup lang="ts">
import type { ISectionField } from "~/interfaces/types";
const props = defineProps<{
  field: ISectionField;
}>();

const longestRowIndex = computed(() => {
  return props.field.value.reduce(
    (maxIndex: number, row: Object, currentIndex: number) =>
      Object.keys(row).length > Object.keys(maxIndex).length
        ? currentIndex
        : maxIndex,
    0
  );
});

function isEmpty(obj: any) {
  for (const prop in obj) {
    if (Object.hasOwn(obj, prop)) {
      return false;
    }
  }

  return true;
}

const rows = computed(() => {
  return props.field.value.map((row: Object) => {
    return Object.entries(row).map(([key, value]) => {
      const filteredValue =
        typeof value === "object"
          ? isEmpty(value)
            ? ""
            : JSON.stringify(value)
          : value;
      return { key, value: filteredValue };
    });
  });
});
</script>

<template>
  <Table>
    <template #head>
      <TableHeadRow>
        <TableHead v-for="longestRowColumns in rows[longestRowIndex]">{{
          longestRowColumns.key
        }}</TableHead>
      </TableHeadRow>
    </template>
    <template #body>
      <TableRow v-for="row in rows">
        <TableCell v-for="longestRowColumns in rows[longestRowIndex]">{{
          row.find((col: any) => col.key === longestRowColumns.key)?.value
        }}</TableCell>
      </TableRow>
    </template>
  </Table>
</template>
