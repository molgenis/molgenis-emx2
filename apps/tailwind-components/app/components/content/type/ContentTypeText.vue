<script setup lang="ts">
import type { ISectionField } from "../../../../types/types";
import ContentReadMore from "../../ContentReadMore.vue";

defineProps<{
  field: ISectionField;
}>();
</script>

<template>
  <ContentReadMore :text="field.value" :cutoff="500" />
</template>
