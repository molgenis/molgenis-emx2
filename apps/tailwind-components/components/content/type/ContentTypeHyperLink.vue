<script setup lang="ts">
import type { ISectionField } from "~/interfaces/types";

defineProps<{
  field: ISectionField;
}>();
</script>

<template>
  <HyperLink :href="field.value" target="_blank"></HyperLink>
</template>
