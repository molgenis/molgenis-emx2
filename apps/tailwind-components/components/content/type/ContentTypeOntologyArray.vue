<script setup lang="ts">
import type { ISectionField, IOntologyNode } from "~/interfaces/types";
const { field } = defineProps<{
  field: ISectionField;
}>();

const asString = computed(() => {
  return field.value?.map((item): any => item.name).join(", ");
});
</script>

<template>
  <p>{{ asString }}</p>
</template>
