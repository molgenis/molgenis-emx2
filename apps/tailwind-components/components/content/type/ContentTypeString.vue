<script setup lang="ts">
import type { ISectionField } from "~/interfaces/types";
defineProps<{
  field: ISectionField;
}>();
</script>

<template>
  <p>{{ field.value }}</p>
</template>
