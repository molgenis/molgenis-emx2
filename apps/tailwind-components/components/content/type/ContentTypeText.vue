<script setup lang="ts">
import type { ISectionField } from "../../../types/types";

defineProps<{
  field: ISectionField;
}>();
</script>

<template>
  <ContentReadMore :text="field.value" :cutoff="500" />
</template>
