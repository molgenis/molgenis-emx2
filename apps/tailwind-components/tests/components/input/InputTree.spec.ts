import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

import InputTree from "@/components/input/Tree.vue";
import type { ITreeNode } from "~/types/types";

function generateTreeData(width: number, depth: number, parentName?: string) {
  const nodes = [];
  for (let i = 0; i < width; i++) {
    const name = parentName ? parentName + `.${i}` : `Node ${i}`;
    const node: ITreeNode = {
      name,
      selected: false,
      expanded: false,
      children: depth > 0 ? generateTreeData(width, depth - 1, name) : [],
    };

    nodes.push(node);
  }
  return nodes;
}

describe("InputTree", () => {
  it("show top level nodes as list by default ( tree is collapes)", () => {
    const width = 2;
    const depth = 4;
    const rootNodes = generateTreeData(width, depth);
    const wrapper = mount(InputTree, {
      props: {
        rootNodes,
        modelValue: [],
      },
    });
    expect(wrapper.find("ul").exists()).toBe(true);
    // 3 ul 's for depth of 4, but we only get 3 (and a <treechild> stub) because of shallow mount, stange
    expect(wrapper.find("ul > li > ul > li > ul ").exists()).toBe(true);
    expect(wrapper.findAll("ul:first-child > li").length).toEqual(width);

    // now try using a snapshot
    expect(wrapper.element).toMatchSnapshot();
  });
});
