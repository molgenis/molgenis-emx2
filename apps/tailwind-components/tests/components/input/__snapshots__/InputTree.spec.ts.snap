// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputTree > show top level nodes as list by default ( tree is collapes) 1`] = `
<ul
  class="text-search-filter-group-title"
>
  
  <li
    class="mb-2.5"
  >
    <div
      class="flex items-start"
    >
      <span
        class="flex items-center justify-center w-6 h-6 rounded-full hover:bg-search-filter-group-toggle hover:cursor-pointer rotate-180 text-search-filter-group-toggle"
      >
        <iconscaretup
          width="20"
        />
      </span>
      <div
        class="flex items-center"
      >
        <input
          class="w-5 h-5 rounded-3px ml-[6px] mr-2.5 mt-0.5 accent-yellow-500 indeterminate:accent-yellow-500 border border-checkbox hover:cursor-pointer"
          id="Node 0"
          name="Node 0"
          type="checkbox"
        />
      </div>
      <label
        class="hover:cursor-pointer text-body-sm group"
        for="Node 0"
      >
        <span
          class="group-hover:underline whitespace-nowrap"
        >
          Node 0
        </span>
      </label>
      <div
        class="inline-flex items-center whitespace-nowrap"
      >
        <div
          class="inline-block pl-1"
        >
          <!--v-if-->
        </div>
      </div>
    </div>
    <ul
      class="ml-10 mr-4 hidden"
    >
      
      <li
        class="mt-2.5 relative"
      >
        <span
          class="flex items-center"
        >
          <span
            class="-left-[11px] top-0 text-search-filter-group-toggle rounded-full hover:bg-search-filter-group-toggle hover:cursor-pointer h-6 w-6 flex items-center justify-center absolute z-20"
          >
            <iconscaretup
              width="20"
            />
          </span>
          <iconscollapsiblelistitemsub
            class="text-blue-200 absolute -top-[9px]"
            width="20"
          />
        </span>
        <div
          class="flex items-start ml-3"
        >
          <div
            class="flex items-center"
          >
            <input
              class="w-5 h-5 rounded-3px ml-2.5 mr-2.5 mt-0.5 accent-yellow-500 indeterminate:accent-yellow-500 border border-checkbox"
              id="Node 0.0"
              name="Node 0.0"
              type="checkbox"
            />
          </div>
          <label
            class="hover:cursor-pointer text-body-sm group"
            for="Node 0.0"
          >
            <span
              class="group-hover:underline whitespace-nowrap"
            >
              Node 0.0
            </span>
            <div
              class="inline-flex items-center whitespace-nowrap"
            >
              <div
                class="inline-block"
              >
                <!--v-if-->
              </div>
            </div>
          </label>
        </div>
        <ul
          class="hidden ml-[31px]"
        >
          <treechild
            expandselected="false"
            nodes="[object Object],[object Object]"
            parent="[object Object]"
          />
        </ul>
      </li>
      <li
        class="mt-2.5 relative"
      >
        <span
          class="flex items-center"
        >
          <span
            class="-left-[11px] top-0 text-search-filter-group-toggle rounded-full hover:bg-search-filter-group-toggle hover:cursor-pointer h-6 w-6 flex items-center justify-center absolute z-20"
          >
            <iconscaretup
              width="20"
            />
          </span>
          <iconscollapsiblelistitemsub
            class="text-blue-200 absolute -top-[9px]"
            width="20"
          />
        </span>
        <div
          class="flex items-start ml-3"
        >
          <div
            class="flex items-center"
          >
            <input
              class="w-5 h-5 rounded-3px ml-2.5 mr-2.5 mt-0.5 accent-yellow-500 indeterminate:accent-yellow-500 border border-checkbox"
              id="Node 0.1"
              name="Node 0.1"
              type="checkbox"
            />
          </div>
          <label
            class="hover:cursor-pointer text-body-sm group"
            for="Node 0.1"
          >
            <span
              class="group-hover:underline whitespace-nowrap"
            >
              Node 0.1
            </span>
            <div
              class="inline-flex items-center whitespace-nowrap"
            >
              <div
                class="inline-block"
              >
                <!--v-if-->
              </div>
            </div>
          </label>
        </div>
        <ul
          class="hidden ml-[31px]"
        >
          <treechild
            expandselected="false"
            nodes="[object Object],[object Object]"
            parent="[object Object]"
          />
        </ul>
      </li>
      
    </ul>
  </li>
  <li
    class="mb-2.5"
  >
    <div
      class="flex items-start"
    >
      <span
        class="flex items-center justify-center w-6 h-6 rounded-full hover:bg-search-filter-group-toggle hover:cursor-pointer rotate-180 text-search-filter-group-toggle"
      >
        <iconscaretup
          width="20"
        />
      </span>
      <div
        class="flex items-center"
      >
        <input
          class="w-5 h-5 rounded-3px ml-[6px] mr-2.5 mt-0.5 accent-yellow-500 indeterminate:accent-yellow-500 border border-checkbox hover:cursor-pointer"
          id="Node 1"
          name="Node 1"
          type="checkbox"
        />
      </div>
      <label
        class="hover:cursor-pointer text-body-sm group"
        for="Node 1"
      >
        <span
          class="group-hover:underline whitespace-nowrap"
        >
          Node 1
        </span>
      </label>
      <div
        class="inline-flex items-center whitespace-nowrap"
      >
        <div
          class="inline-block pl-1"
        >
          <!--v-if-->
        </div>
      </div>
    </div>
    <ul
      class="ml-10 mr-4 hidden"
    >
      
      <li
        class="mt-2.5 relative"
      >
        <span
          class="flex items-center"
        >
          <span
            class="-left-[11px] top-0 text-search-filter-group-toggle rounded-full hover:bg-search-filter-group-toggle hover:cursor-pointer h-6 w-6 flex items-center justify-center absolute z-20"
          >
            <iconscaretup
              width="20"
            />
          </span>
          <iconscollapsiblelistitemsub
            class="text-blue-200 absolute -top-[9px]"
            width="20"
          />
        </span>
        <div
          class="flex items-start ml-3"
        >
          <div
            class="flex items-center"
          >
            <input
              class="w-5 h-5 rounded-3px ml-2.5 mr-2.5 mt-0.5 accent-yellow-500 indeterminate:accent-yellow-500 border border-checkbox"
              id="Node 1.0"
              name="Node 1.0"
              type="checkbox"
            />
          </div>
          <label
            class="hover:cursor-pointer text-body-sm group"
            for="Node 1.0"
          >
            <span
              class="group-hover:underline whitespace-nowrap"
            >
              Node 1.0
            </span>
            <div
              class="inline-flex items-center whitespace-nowrap"
            >
              <div
                class="inline-block"
              >
                <!--v-if-->
              </div>
            </div>
          </label>
        </div>
        <ul
          class="hidden ml-[31px]"
        >
          <treechild
            expandselected="false"
            nodes="[object Object],[object Object]"
            parent="[object Object]"
          />
        </ul>
      </li>
      <li
        class="mt-2.5 relative"
      >
        <span
          class="flex items-center"
        >
          <span
            class="-left-[11px] top-0 text-search-filter-group-toggle rounded-full hover:bg-search-filter-group-toggle hover:cursor-pointer h-6 w-6 flex items-center justify-center absolute z-20"
          >
            <iconscaretup
              width="20"
            />
          </span>
          <iconscollapsiblelistitemsub
            class="text-blue-200 absolute -top-[9px]"
            width="20"
          />
        </span>
        <div
          class="flex items-start ml-3"
        >
          <div
            class="flex items-center"
          >
            <input
              class="w-5 h-5 rounded-3px ml-2.5 mr-2.5 mt-0.5 accent-yellow-500 indeterminate:accent-yellow-500 border border-checkbox"
              id="Node 1.1"
              name="Node 1.1"
              type="checkbox"
            />
          </div>
          <label
            class="hover:cursor-pointer text-body-sm group"
            for="Node 1.1"
          >
            <span
              class="group-hover:underline whitespace-nowrap"
            >
              Node 1.1
            </span>
            <div
              class="inline-flex items-center whitespace-nowrap"
            >
              <div
                class="inline-block"
              >
                <!--v-if-->
              </div>
            </div>
          </label>
        </div>
        <ul
          class="hidden ml-[31px]"
        >
          <treechild
            expandselected="false"
            nodes="[object Object],[object Object]"
            parent="[object Object]"
          />
        </ul>
      </li>
      
    </ul>
  </li>
  
</ul>
`;
